cmd/thanos/rule.go

```diff
@@ -963,3 +1035,88 @@ func validateTemplate(tmplStr string) error {
 	}
 	return nil
 }
+
+func prepareEndpointSet(
```

Comment:
This looks like repeated code with the querier. Is there a chance to use a common function?

Reply:
I moved it to query and extended it, so it's used by both query and rule now.